### DISCUSSION TITLE
are NSCZ and NSCR from v.2 available in v.4?

### DISCUSSION THREAD

**Discussion by ppugpug** [svenreiche/Genesis-1.3-Version4#83] [Q&A] [answered]
For THz FEL, the momentum of an electron beam is low, so space-charge effects play a significant role. Such effects can partially be implemented in Genesis 1.3 version 2 using input parameters NSCZ and NSCR.  

Are these input parameters available in Genesis 1.3 version 4? If not, will developers plan to implement them in the "near" future?

**Comment by svenreiche** [comment, marked as answer]
I have it on my list to implement. The code is there (imported from version 2) but the structure of the core algorithm has change and it is not that straight forward to implement it. I hope that I have some time to work on Genesis this summer.
